### ISSUE TITLE
InnerAssignment and Lambda Expressions

### ISSUE THREAD

**Issue by EvanCahill** [checkstyle/checkstyle#377]
The `InnerAssignment` check incorrectly reports assignment statements in a single statement lambda expression as inner assignments:

``` java
// Causes an "avoid inner assignment" checkstyle error
button.setOnAction(e -> pressed = true);

// Does not cause an error
button.setOnAction(e -> { pressed = true; });
```
